build/template-OneBranch-CI-libsandsamples.yaml

```
@@ -21,49 +22,82 @@ steps:
   inputs:
     version: 6.x
 
-- task: DotNetCoreCLI@2
-  displayName: 'dotnet workload restore for Desktop project'
+# Compute the source directory based on the parameter MsalSourceDir
+- task: PowerShell@2
+  displayName: 'Compute Source Directory'
   inputs:
-    command: 'custom'
-    custom: 'workload'
-    arguments: 'restore microsoft-authentication-library-for-dotnet\src\client\Microsoft.Identity.Client.Desktop\Microsoft.Identity.Client.Desktop.csproj'
+    targetType: 'inline'
+    script: |
+      # Compute base directory
+      $MsalBaseDir = "$(Build.SourcesDirectory)/$(MsalSourceDir)"
+      
+      # Check if both projects exist under MsalSourceDir
+      $BrokerExists = Test-Path "$MsalBaseDir/src/client/Microsoft.Identity.Client.Broker"
+      $ClientExists = Test-Path "$MsalBaseDir/src/client/Microsoft.Identity.Client"
+      
+      # Check if both projects exist directly under Build.SourcesDirectory
+      $AltBrokerExists = Test-Path "$(Build.SourcesDirectory)/src/client/Microsoft.Identity.Client.Broker"
+      $AltClientExists = Test-Path "$(Build.SourcesDirectory)/src/client/Microsoft.Identity.Client"
+
+      if ($BrokerExists -and $ClientExists) {
+          Write-Host "##vso[task.setvariable variable=SourceDir]$MsalBaseDir"
+      } elseif ($AltBrokerExists -and $AltClientExists) {
+          Write-Host "##vso[task.setvariable variable=SourceDir]$(Build.SourcesDirectory)/"
+      } else {
+          Write-Error "Unable to determine the correct source directory structure."
+          exit 1
+      }
 
 - task: DotNetCoreCLI@2
-  displayName: 'dotnet workload restore for Identity.Client'
+  displayName: 'Restore required workloads for projects'
   inputs:
     command: 'custom'
     custom: 'workload'
-    arguments: 'restore microsoft-authentication-library-for-dotnet\src\client\Microsoft.Identity.Client\Microsoft.Identity.Client.csproj'
+    arguments: >
+      restore $(SourceDir)src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj
+      restore $(SourceDir)src/client/Microsoft.Identity.Client.Broker/Microsoft.Identity.Client.Broker.csproj
+      --configfile $(SourceDir)NuGet.config
```

```suggestion
      --configfile $(SourceDir)NuGet.config
```

@bgavrilMS @neha-bhargava I needed to add this line so it restore points to internal feeds. I was able to test this with KS3. I will document KS3 and how to test using KS3